src/logger/logger.service.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  Injectable,
+  Logger as NestLogger,
+  LoggerService,
+  Scope,
+} from '@nestjs/common';
+import {
+  createLogger,
+  format,
+  Logger as WinstonLogger,
+  transports,
+} from 'winston';
+
+import { ConfigService } from '@nestjs/config';
+
+@Injectable({ scope: Scope.TRANSIENT })
```

Comment:
I think there's probably an unwritten rule against allowing enhancers to be REQUEST scoped. I remember I had issues in the past with this and the service was returning undefined because i changed it to REQUEST scope but you're right there isn't any reason in principle why it shouldn't be shared across consumers.